doc/architecture.md

```diff
@@ -0,0 +1,96 @@
+# Architecture
+
+The architecture of the system tests consist out of two layers.
+The interface between the layers are docker images.
+
+## Grammar
+
+```
+// terminals
+System         = string
+TestClass      = string
+Feature        = string
+Dockerfile     = "Dockerfile"
+TestPrefix     = "Test"
+.              = "."
+_              = "_"
+
+// Non-terminals
+SystemSpec     = System [. Feature]*
+Base           = Dockerfile . SystemSpec
+TestSpec       = TestClass | TestClass SystemSpec
+Test           = TestPrefix _ TestSpec
```

Comment:
Should't this be:
```
TestSpec       = TestClass | TestClass.SystemSpec
Test           = TestPrefix_TestSpec
```
(removed white space, add period)

Reply:
The whitespace is only in the grammar to separate different parts, it is not it the result. The examples should clarify this.

The period was missing. Thanks!